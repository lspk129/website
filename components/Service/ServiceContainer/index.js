import styled from 'styled-components';

const ServiceContainer = styled.div`
  text-align: center;
  display: grid;
  grid-template-columns: 375px 375px;
  grid-template-rows: 260px;
  padding-bottom: 100px;
  justify-content: center;
  align-content: center;
  
`;

export default ServiceContainer;
