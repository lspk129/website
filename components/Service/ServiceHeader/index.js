import styled from 'styled-components';

const ServiceHeader = styled.h2`
  text-align: center;
  padding: 60px;
  margin: 0;
`;

export default ServiceHeader;
