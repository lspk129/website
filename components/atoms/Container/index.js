import styled from 'styled-components';

const Container = styled.div`
  ${({ display }) => display && `display: ${display}`}
  ${({ textAlign }) => textAlign && `text-align: ${textAlign}`};
  ${({ alignSelf }) => alignSelf && `align-self: ${alignSelf}`};
  ${({ alignItems }) => alignItems && `align-items: ${alignItems}`};

  /* margin properties */
  ${({ m }) => m && `margin: ${m}rem;`}
  ${({ mx }) => mx && `margin: 0rem ${mx}rem;`}
  ${({ my }) => my && `margin: ${my}rem 0rem;`}
  ${({ mt }) => mt && `margin-top: ${mt}rem;`}
  ${({ mb }) => mb && `margin-bottom: ${mb}rem;`}
  ${({ mr }) => mr && `margin-right: ${mr}rem;`}
  ${({ ml }) => ml && `margin-left: ${ml}rem;`}

  /* padding properties */
  ${({ p }) => p && `padding: ${p}rem;`}
  ${({ px }) => px && `padding: 0rem ${px}rem;`}
  ${({ py }) => py && `padding: ${py}rem 0rem;`}
  ${({ pt }) => pt && `padding-top: ${pt}rem;`}
  ${({ pb }) => pb && `padding-bottom: ${pb}rem;`}
  ${({ pr }) => pr && `padding-right: ${pr}rem;`}
  ${({ pl }) => pl && `padding-left: ${pl}rem;`}

  /* background properties */
  ${({ bg, src }) => bg && `
    background-image: url(${src});
    background-repeat: no-repeat;
    background-attachment: fixed;
    background-size: cover;
    display: grid;
  `}
  ${({ backgroundColor }) => backgroundColor && `
    background-color: ${backgroundColor};
  `}
  ${({ zIndex }) => zIndex && `z-index: ${zIndex};`}
`;

export default Container;
