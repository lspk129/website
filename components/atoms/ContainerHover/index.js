import styled from 'styled-components';

const ContainerHover = styled.div`
  transition: all 0.5s ease-in-out;
  ${({ scale }) => scale && `
    img {
      width: 200px;
      height: 200px;
    }
    &:hover {
      transform: scale(${scale});
    }
  `}
  .icons {
    transition: all 0.5s ease-in-out;
    color: ${({ theme }) => theme.background};
  }
  ${({ backgroundColor, theme }) => backgroundColor && `
    &:hover {
      background: ${theme.background};
      color: ${theme.text};
      .icons {
        color: ${theme.text};
      }
    };
  `}
`;

export default ContainerHover;
