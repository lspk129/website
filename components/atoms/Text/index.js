import { createElement } from 'react';
import styled from 'styled-components';

const Text = styled(({ tag, hero, upperCase, children, ...props }) => createElement(tag, props, children))`
  text-align: center;

  ${({ tag, theme }) => tag === 'a' && `
    text-decoration: none;
    cursor: pointer;
    color: inherit;
    &:hover { color: ${theme.main}}
  `}

  ${({ tag, hero, theme }) => tag === 'h1' && `
    ${hero && `
      color: ${theme.text};
      font-size: 4rem;
      white-space: nowrap;
      font-weight: bold;
      text-shadow: 0 0.1rem 0.1rem rgba(0, 0, 0, 0.4);
      line-height: 4rem;
    `}
  `}

  ${({ tag, hero, theme }) => tag === 'h2' && `
    ${hero && `
      color: ${theme.text};
      text-shadow: 0 0.1rem 0.1rem rgba(0, 0, 0, 0.2);
    `}
  `}

  ${({ upperCase }) => upperCase && 'text-transform: uppercase'};
`;

Text.defaultProps = {
  tag: 'span',
};

export default Text;
