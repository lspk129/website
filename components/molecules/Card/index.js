import React from 'react';
import PropTypes from 'prop-types';

import Division from '../../atoms/Division';
import Title from '../../atoms/Title';

const Card = ({ image, label, text }) => (
  <Division textAlign={'center'} py={3} pr={1} pl={1}>
    {image}
    <Title tag={'h3'}>{label}</Title>
    <Title tag={'p'}>{text}</Title>
  </Division>
);

Card.propTypes = {
  image: PropTypes.node,
  label: PropTypes.string,
  text: PropTypes.string,
};

Card.defaultProps = {
  image: '',
  label: '',
  text: '',
};

export default Card;
