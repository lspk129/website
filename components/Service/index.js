import React from 'react';

import ServiceHeader from './ServiceHeader';
import ServiceContainer from './ServiceContainer';
import ServiceItem from './ServiceItem';

const Service = () => (
  <div>
    <ServiceHeader>
      What we do
    </ServiceHeader>
    <ServiceContainer>
      <ServiceItem>
        <i className="material-icons">desktop_mac</i>
        <h3>Web development</h3>
        <p>We help you build fast, functional, performance oriented websites and web apps</p>
      </ServiceItem>
      <ServiceItem>
        <i className="material-icons">color_lens</i>
        <h3>Web design</h3>
        <p>We are creating unique and adaptive designs to engage customers</p>
      </ServiceItem>
    </ServiceContainer>
  </div>
);

export default Service;
