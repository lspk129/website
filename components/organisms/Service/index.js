import React from 'react';
import { MdDesktopMac, MdColorLens } from 'react-icons/lib/md';
import { Row, Col } from 'react-styled-flexboxgrid';

import Card from '../../molecules/Card';
import Division from '../../atoms/Division';
import DivisionHover from '../../atoms/DivisionHover';
import Text from '../../atoms/Text';

const Service = () => (
  <Division bg src={'../../../static/Service_image.jpg'} pb={3}>
    <Division py={3}>
      <Text tag={'h2'}>What we do</Text>
    </Division>
    <Row center={'xs'}>
      <Col md={3} sm={4} xs={12}>
        <DivisionHover backgroundColor>
          <Card
            image={<MdDesktopMac className={'icons'} size={48} />}
            label={'Web development'}
            text={'We help you build fast, functional, performance oriented websites and web apps'}
          />
        </DivisionHover>
      </Col>
      <Col md={3} sm={4} xs={12}>
        <DivisionHover backgroundColor>
          <Card
            image={<MdColorLens className={'icons'} size={48} />}
            label={'Web design'}
            text={'We are creating unique and adaptive designs to engage customers'}
          />
        </DivisionHover>
      </Col>
    </Row>
  </Division>
);

export default Service;
