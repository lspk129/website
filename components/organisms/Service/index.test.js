import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MdDesktopMac, MdColorLens } from 'react-icons/lib/md';

import Service from './';
import Container from '../../atoms/Container';
import ContainerHover from '../../atoms/ContainerHover';
import Card from '../../molecules/Card';
import Text from '../../atoms/Text';

describe('Service', () => {
  const wrapper = mount(<Service />);

  it('renders correctly', () => {
    expect(wrapper).toBePresent();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Text component', () => {
    expect(wrapper).toBePresent();
    expect(wrapper.find(Text)).toBePresent();
  });

  it('renders text "What we do"', () => {
    expect(wrapper.find(Text).at(0)).toHaveText('What we do');
  });
  it('renders Card component', () => {
    expect(wrapper.find(Card)).toBePresent();
  });

  it('renders Container component', () => {
    expect(wrapper.find(Container)).toBePresent();
  });

  it('renders ContainerHover component', () => {
    expect(wrapper.find(ContainerHover)).toBePresent();
  });

  it('renders Card component with text "Web development"', () => {
    expect(wrapper.find(Card).at(0)).toHaveProp('label', 'Web development');
  });

  it('renders Card component with content "We help you build fast, functional, performance oriented websites and web apps"', () => {
    expect(wrapper.find(Card).at(0)).toHaveProp(
      'text',
      'We help you build fast, functional, performance oriented websites and web apps',
    );
  });

  it('renders Card component with text "Web design"', () => {
    expect(wrapper.find(Card).at(1)).toHaveProp('label', 'Web design');
  });

  it('renders Card component with content "We are creating unique and adaptive designs to engage customers"', () => {
    expect(wrapper.find(Card).at(1)).toHaveProp(
      'text',
      'We are creating unique and adaptive designs to engage customers',
    );
  });

  it('renders MdDesktopMac component with desktop icon', () => {
    expect(wrapper.find(MdDesktopMac)).toBePresent();
  });

  it('renders MdColorLens component with color lens icon', () => {
    expect(wrapper.find(MdColorLens)).toBePresent();
  });
});
