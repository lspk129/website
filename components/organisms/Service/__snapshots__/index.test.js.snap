// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Service renders correctly 1`] = `
<Service>
  <styled.div
    bg={true}
    pb={3}
    src="../../../static/Service_image.jpg"
  >
    <div
      className="sc-bdVaJa eAJuBK"
      src="../../../static/Service_image.jpg"
    >
      <styled.div
        py={3}
      >
        <div
          className="sc-bdVaJa gnRkQa"
        >
          <Styled(Component)
            tag="h2"
          >
            <Component
              className="sc-bwzfXH fbaTre"
              tag="h2"
            >
              <h2
                className="sc-bwzfXH fbaTre"
              >
                What we do
              </h2>
            </Component>
          </Styled(Component)>
        </div>
      </styled.div>
      <Row
        center="xs"
      >
        <div
          className="Row__Row-kNtYPA jaQNpG"
        >
          <Col
            md={3}
            sm={4}
            xs={12}
          >
            <div
              className="Col__Col-hCQanm fyLqBb"
            >
              <styled.div
                backgroundColor={true}
              >
                <div
                  className="sc-htpNat ghwXNe"
                >
                  <Card
                    image={
                      <MdDesktopMac
                        className="icons"
                        size={48}
                      />
                    }
                    label="Web development"
                    text="We help you build fast, functional, performance oriented websites and web apps"
                  >
                    <styled.div
                      pl={1}
                      pr={1}
                      py={3}
                      textAlign="center"
                    >
                      <div
                        className="sc-bdVaJa khboWb"
                      >
                        <MdDesktopMac
                          className="icons"
                          size={48}
                        >
                          <IconBase
                            className="icons"
                            size={48}
                            viewBox="0 0 40 40"
                          >
                            <svg
                              className="icons"
                              fill="currentColor"
                              height={48}
                              preserveAspectRatio="xMidYMid meet"
                              style={
                                Object {
                                  "color": undefined,
                                  "verticalAlign": "middle",
                                }
                              }
                              viewBox="0 0 40 40"
                              width={48}
                            >
                              <g>
                                <path
                                  d="m35 23.4v-16.8h-30v16.8h30z m0-20c1.8 0 3.4 1.4 3.4 3.2v20c0 1.8-1.6 3.4-3.4 3.4h-11.6l3.2 5v1.6h-13.2v-1.6l3.2-5h-11.6c-1.8 0-3.4-1.6-3.4-3.4v-20c0-1.8 1.6-3.2 3.4-3.2h30z"
                                />
                              </g>
                            </svg>
                          </IconBase>
                        </MdDesktopMac>
                        <Styled(Component)
                          tag="h3"
                        >
                          <Component
                            className="sc-bwzfXH elPfnD"
                            tag="h3"
                          >
                            <h3
                              className="sc-bwzfXH elPfnD"
                            >
                              Web development
                            </h3>
                          </Component>
                        </Styled(Component)>
                        <Styled(Component)
                          tag="p"
                        >
                          <Component
                            className="sc-bwzfXH elPfnD"
                            tag="p"
                          >
                            <p
                              className="sc-bwzfXH elPfnD"
                            >
                              We help you build fast, functional, performance oriented websites and web apps
                            </p>
                          </Component>
                        </Styled(Component)>
                      </div>
                    </styled.div>
                  </Card>
                </div>
              </styled.div>
            </div>
          </Col>
          <Col
            md={3}
            sm={4}
            xs={12}
          >
            <div
              className="Col__Col-hCQanm fyLqBb"
            >
              <styled.div
                backgroundColor={true}
              >
                <div
                  className="sc-htpNat ghwXNe"
                >
                  <Card
                    image={
                      <MdColorLens
                        className="icons"
                        size={48}
                      />
                    }
                    label="Web design"
                    text="We are creating unique and adaptive designs to engage customers"
                  >
                    <styled.div
                      pl={1}
                      pr={1}
                      py={3}
                      textAlign="center"
                    >
                      <div
                        className="sc-bdVaJa khboWb"
                      >
                        <MdColorLens
                          className="icons"
                          size={48}
                        >
                          <IconBase
                            className="icons"
                            size={48}
                            viewBox="0 0 40 40"
                          >
                            <svg
                              className="icons"
                              fill="currentColor"
                              height={48}
                              preserveAspectRatio="xMidYMid meet"
                              style={
                                Object {
                                  "color": undefined,
                                  "verticalAlign": "middle",
                                }
                              }
                              viewBox="0 0 40 40"
                              width={48}
                            >
                              <g>
                                <path
                                  d="m29.1 20q1.1 0 1.8-0.7t0.7-1.8-0.7-1.8-1.8-0.7-1.7 0.7-0.8 1.8 0.8 1.8 1.7 0.7z m-5-6.6q1.1 0 1.8-0.8t0.7-1.7-0.7-1.8-1.8-0.7-1.7 0.7-0.8 1.8 0.8 1.7 1.7 0.8z m-8.2 0q1 0 1.7-0.8t0.8-1.7-0.8-1.8-1.7-0.7-1.8 0.7-0.7 1.8 0.7 1.7 1.8 0.8z m-5 6.6q1 0 1.7-0.7t0.8-1.8-0.8-1.8-1.7-0.7-1.8 0.7-0.7 1.8 0.7 1.8 1.8 0.7z m9.1-15q6.2 0 10.6 3.9t4.4 9.5q0 3.4-2.5 5.8t-5.9 2.4h-2.8q-1.1 0-1.8 0.8t-0.7 1.7q0 0.9 0.6 1.7t0.6 1.7q0 1.1-0.7 1.8t-1.8 0.7q-6.2 0-10.6-4.4t-4.4-10.6 4.4-10.6 10.6-4.4z"
                                />
                              </g>
                            </svg>
                          </IconBase>
                        </MdColorLens>
                        <Styled(Component)
                          tag="h3"
                        >
                          <Component
                            className="sc-bwzfXH elPfnD"
                            tag="h3"
                          >
                            <h3
                              className="sc-bwzfXH elPfnD"
                            >
                              Web design
                            </h3>
                          </Component>
                        </Styled(Component)>
                        <Styled(Component)
                          tag="p"
                        >
                          <Component
                            className="sc-bwzfXH elPfnD"
                            tag="p"
                          >
                            <p
                              className="sc-bwzfXH elPfnD"
                            >
                              We are creating unique and adaptive designs to engage customers
                            </p>
                          </Component>
                        </Styled(Component)>
                      </div>
                    </styled.div>
                  </Card>
                </div>
              </styled.div>
            </div>
          </Col>
        </div>
      </Row>
    </div>
  </styled.div>
</Service>
`;
